### DISCUSSION TITLE
value_and_grad for vector-valued functions

### DISCUSSION THREAD

**Discussion by Ueaj-Kerman** [jax-ml/jax#16959] [General] [answered]
`jax.jacfwd` works in place of `jax.grad` for vector-valued functions, however there doesn't seem to be an analog of `jax.value_and_grad` for vector-valued functions.

Is there such an equivalent, or is it not applicable to this scenario due to the intricacies of auto differentiation?

**Comment by jakevdp** [comment, marked as answer]
Hi - thanks for the question!

See https://github.com/google/jax/pull/762#issuecomment-1001885437 for some discussion of why this isn't included in JAX. The thread below that comment has some suggestions of how you might compute this in cases where you might need it.
